### PR TITLE
Sam/doc size limit

### PR DESCRIPTION
**Pull request by sam0044**
### Motivation and Context

Create a default specialization to revert to incase of no specialization.

### Description

This PR adds a checkbox that can be used to set a specialization as a default specialization. When there is no specialization set, it will throw a warning that having a default specialization is a requirement. Once you set the default and try to set another specialization as default, it will throw a confirmation dialog prompting the user to confirm the change. I also added the deployment field requirement as part of this PR. After you set the default specialization, you should be able to chat or start a chat by uploading a document. In the case of document upload starting the chat,  because when we set the specialization the chat state changes and the spinner next to attachment button goes away, for the sole case of starting a chat with document upload, I added a info alert notifying the user that document is still being processed. I could add the global spinner that just loads until the doc is uploaded. Whichever is preferred or if someone has a better suggestion. For specialization deletion I made it so that you can't delete the default one if there's other specializations present. If you try to do so, the system informs you to set another specialization as default. You can delete the default specialization when its the only one left. P.S I added a confirmation dialog for deleting a specialization as well. Added logic for when you try to toggle the default specialization, it won't let you.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: webapi/Models/Response/QSpecializationResponse.cs [hunk 1]
```diff
@@ -77,6 +77,12 @@ public class QSpecializationResponse
     [JsonPropertyName("isActive")]
     public bool isActive { get; set; } = true;
 
+    /// <summary>
+    /// Is Default flag of the specialization
+    /// </summary>
+    [JsonPropertyName("isDefault")]
+    public bool? IsDefault { get; set; } = false;
```
Comment (GrahamS-Quartech) [inline webapi/Models/Response/QSpecializationResponse.cs:84]: Not sure this should have a default value, see: https://github.com/Quartech/chat-copilot/pull/172

**Review by GrahamS-Quartech — APPROVED** [review]
Worked well for me. Adding confirmation to the delete button here is a good change imo. Does not necessarily need to be part of this PR, but I'm just observing that deleting a specialization and then attempting to send another message to a conversation that used that spec raises an error.

File: webapi/Models/Response/QSpecializationResponse.cs (hunk 1 above)
Comment (sam0044) [reply webapi/Models/Response/QSpecializationResponse.cs:84]: from what I remember someone is working on a ticket that deletes the chats for the specializations that are removed. I am sure I saw a ticket for that so I am hoping that ticket would fix that. I'll take a look at the other thing

Comment (sam0044) [reply webapi/Models/Response/QSpecializationResponse.cs:84]: I removed the default value

**Comment by hansonchun-quartech** [comment]
Setting another specialization as the default doesn't seems to work

https://github.com/user-attachments/assets/b209b31e-c422-4a77-a406-16be2fdd5c2f

**Comment by sam0044** [comment]
@hansonchun-quartech you have to click save for it to be actually set. Thats the normal behavior for updating the specialization so I left it that way

**Review by hansonchun-quartech — DISMISSED** [review]
Functionality works well! Tested on my local.

One thing I wonder is if the prompt to change the default specialization after clicking the checkbox is necessary. The prompt seems to suggest that the default specialization will change after clicking "yes", but in actuality you need to save the specialization again to update it.

**Comment by sam0044** [comment]
@hansonchun-quartech had a chat with graham, I might try to make it so that on saving in the confirmation dialog, it updates it. either that or I might change the confirmation dialog to an alert notifying the user and you are wanting to change the default specialization.

**Review by hansonchun-quartech — APPROVED** [review]

**Pull request merged**